src/pages/Login/Login.js

```diff
@@ -1,10 +1,112 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate, Link } from 'react-router-dom';
+import './Login.scss';
 
 const Login = () => {
+  const [userInfo, setUserInfo] = useState({
+    email: '',
+    password: '',
+  });
+
+  const navigate = useNavigate();
+
+  const idCondition = userInfo.email.includes(('@', 4) && '.com');
+  const pwCondition = userInfo.password.length >= 5;
+
+  const handleIdInput = e => {
+    setUserInfo({
+      ...userInfo,
+      email: e.target.value,
+    });
+  };
+
+  const handlePasswordInput = e => {
+    setUserInfo({
+      ...userInfo,
+      password: e.target.value,
+    });
+  };
+
+  const onSubmit = () => {
+    alert('submitted');
+    navigate('/main');
+  };
+
+  const onKeyUp = event => {
+    if (event.keyCode === 13) {
+      onSubmit();
+    }
+  };
+
+  const login = e => {
+    e.preventDefault();
+    if (idCondition && pwCondition) {
+      fetch('http://10.58.52.159:8007/users/signin', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json;charset=utf-8' },
+        body: JSON.stringify({
+          email: userInfo.email,
+          password: userInfo.password,
+        }),
+      })
```

Comment:
```suggestion
      fetch('http://10.58.52.159:8007/users/signin', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json;charset=utf-8' },
        body: JSON.stringify({
          email: email,
          password: password,
        }),
      })
```
1. 구조분해할당을 한다. const {email, password} = userInfo;
2. body를 축약시킨다.

Reply:
```suggestion
      fetch('http://10.58.52.159:8007/users/signin', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json;charset=utf-8' },
        body: JSON.stringify({
          email,
          password
        }),
      })
```
JS 객체단축속성명을 사용하면 위와 같이 표현할 수도 있습니다.
key와 value가 같은 경우입니다!